contracts/src/interfaces/IHyperdriveMatchingEngine.sol

```diff
@@ -0,0 +1,168 @@
+    error AlreadyCancelled();
+
+    /// @notice Thrown when an order is already expired.
+    error AlreadyExpired();
+
+    /// @notice Thrown when the destination for the add or remove liquidity
+    ///         options isn't configured to this contract.
+    error InvalidDestination();
+
+    /// @notice Thrown when orders that don't cross are matched.
+    error InvalidMatch();
+
+    /// @notice Thrown when the order type doesn't match the expected type.
+    error InvalidOrderType();
+
+    /// @notice Thrown when an address that didn't create an order tries to
+    ///         cancel it.
+    error InvalidSender();
+
+    /// @notice Thrown when `asBase = false` is used. This implementation is
+    ///         opinionated to keep the implementation simple.
+    error InvalidSettlementAsset();
+
+    /// @notice Thrown when the signature for an order intent doesn't recover to
+    ///         the expected signer address.
+    error InvalidSignature();
+
+    /// @notice Thrown when the long and short orders don't refer to the same
+    ///         Hyperdrive instance.
+    error MismatchedHyperdrive();
+
+    /// @notice Emitted when orders are cancelled
+    event OrdersCancelled(address indexed trader, bytes32[] orderHashes);
+
+    /// @notice Emitted when orders are matched
+    event OrdersMatched(
+        IHyperdrive indexed hyperdrive,
+        bytes32 indexed longOrderHash,
+        bytes32 indexed shortOrderHash,
+        address long,
+        address short
+    );
+
+    /// @notice The type of an order intent.
+    enum OrderType {
+        OpenLong,
+        OpenShort
+    }
+
+    /// @notice The order intent struct that encodes a trader's desire to trade.
+    struct OrderIntent {
+        /// @dev The trader address that will be charged when orders are matched.
+        address trader;
+        /// @dev The Hyperdrive address where the trade will be executed.
+        IHyperdrive hyperdrive;
+        /// @dev The amount to be used in the trade. In the case of `OpenLong`,
+        ///      this is the amount of base to deposit, and in the case of
+        ///      `OpenShort`, this is the amount of bonds to short.
+        uint256 amount;
+        /// @dev The slippage guard to be used in the trade. In the case of
+        ///      `OpenLong`, this is the minimum output in bonds, and in the
+        ///      case of `OpenShort`, this is the maximum deposit in base.
+        uint256 slippageGuard;
+        /// @dev The minimum vault share price. This protects traders against
+        ///      the sudden accrual of negative interest in a yield source.
+        uint256 minVaultSharePrice;
+        /// @dev The options that configure how the trade will be settled.
+        ///      `asBase` is required to be true, the `destination` is the
+        ///      address that receives the long or short position that is
+        ///      purchased, and the extra data is configured for the yield
```

Comment:
It's mentioned in `HyperdriveMatchingEngine.sol` that the `extraData` param is not used, maybe it's worth mentioning here as well? Reading this gives me the impression that `extraData` is being considered.